Region/Histogram.cc

```diff
@@ -13,40 +13,47 @@ Histogram::Histogram(int num_bins, float min_value, float max_value, const std::
 
 Histogram::Histogram(Histogram& h, tbb::split) : _bin_width(h._bin_width), _min_val(h._min_val), _hist(h._hist.size(), 0), _data(h._data) {}
 
-void Histogram::operator()(const tbb::blocked_range<size_t>& r) {
-    std::vector<int> tmp(_hist);
+void Histogram::operator()(const tbb::blocked_range<uint64_t>& r) {
+    std::vector<uint64_t> tmp(_hist);
     for (auto i = r.begin(); i != r.end(); ++i) {
         auto v = _data[i];
         if (std::isnan(v) || std::isinf(v))
             continue;
-        int bin = std::max(std::min((int)((v - _min_val) / _bin_width), (int)_hist.size() - 1), 0);
+        // int bin = std::max(std::min((int)((v - _min_val) / _bin_width), (int)_hist.size() - 1), 0);
+        uint64_t _1 = (v - _min_val) / _bin_width;
+        uint64_t _2 = _hist.size() - 1;
+        uint64_t _3 = _1 < _2 ? _1 : _2;
+        uint64_t bin = _3 >= 0 ? _3 : 0;
```

Comment:
These changes aren't needed, are they? The new code is pretty ugly

Reply:
Fixed